kong/cmd/utils/prefix_handler.lua

```diff
@@ -378,6 +378,54 @@ local function write_process_secrets_file(path, data)
   return true
 end
 
+local function pre_create_lmdb(conf)
+  local prefix = conf.prefix
+  local lmdb_path = conf.lmdb_environment_path or "dbless.lmdb"
+  local user = string.match(conf.nginx_user or "", "%w+") or "`whoami`"
+
+  local dir_name
+  if lmdb_path:sub(1, 1) == "/" then
+    dir_name = lmdb_path
+
+  else
+    dir_name = prefix .. "/" .. lmdb_path
+  end
+
+  if pl_path.isdir(dir_name) then
+    return true
+  end
+
+  log.debug("LMDB directory '%s' does not exist, " ..
+            "pre-creating with the correct permissions",
+            dir_name)
+
+  local ok, err = pl_path.mkdir(dir_name)
+  if not ok then
+    return nil, "can not create directory for LMDB " .. dir_name ..
+                ", err: " .. err
+  end
+
+  local cmds = {
+    string.format("chown %s %s && chmod 0700 %s",
+                  user, dir_name, dir_name),
+
+    string.format("touch %s && chmod 0600 %s",
+                  dir_name .. "/data.mdb", dir_name .. "/data.mdb"),
+
+    string.format("touch %s && chmod 0600 %s",
+                  dir_name .. "/lock.mdb", dir_name .. "/lock.mdb"),
+  }
+
+  for _, cmd in ipairs(cmds) do
+    local ok, _, _, stderr = pl_utils.executeex(cmd)
+    if not ok then
+      return nil, stderr
```

Comment:
This should have a better error message, something like:

`return nil, "unable to set correct permissions for LMDB files: " .. stderr`

Also, we should probably only log an error when these command fails. We can let Kong to continue starting and if these permissions are not working, the worker will complain anyway.

Reply:
if `pre_create_lmdb` fails, we use `log.warn` and continue the workflow.